sds/src/secondary_validation/spain_dni_checksum.rs

```diff
@@ -0,0 +1,75 @@
+use crate::secondary_validation::Validator;
+
+pub struct SpanishDniChecksum;
+
+const LETTER_TABLE: [&str; 23] = [
+    "T", "R", "W", "A", "G", "M", "Y", "F", "P", "D", "X", "B", "N", "J", "Z", "S", "Q", "V", "H",
+    "L", "C", "K", "E",
+];
+
+const DNI_LENGTH: usize = 9;
+const NUMBER_LENGTH: usize = 8;
+impl Validator for SpanishDniChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        /*
+         * Spanish DNI format:
+         * 8 digits followed by a letter
+         * The letter is calculated by taking the number modulo 23 and using it as an index
+         * into LETTER_TABLE
+         */
+        if regex_match.len() != DNI_LENGTH {
+            return false;
+        }
+
+        // Get the number and letter parts
+        let (number_part, letter_part) = regex_match.split_at(8);
+
+        // Check if first 8 chars are digits and last is a letter
+        if !number_part.chars().all(|c| c.is_ascii_digit()) {
+            return false;
+        }
```

Comment:
This check should be done by the regex, we could avoid it here. We could also filter out all separators to make the match validation more flexible (in case the regex contains separators)